Lib/collections/__init__.py

```diff
@@ -454,12 +459,14 @@ def __getnewargs__(self):
     cache = _nt_itemgetters
     for index, name in enumerate(field_names):
         try:
-            itemgetter_object, doc = cache[index]
+            tuplegetter_object, doc = cache[index]
         except KeyError:
-            itemgetter_object = _itemgetter(index)
             doc = f'Alias for field number {index}'
-            cache[index] = itemgetter_object, doc
-        class_namespace[name] = property(itemgetter_object, doc=doc)
+            tuplegetter_object = _tuplegetter(index, doc=doc)
+            cache[index] = tuplegetter_object, doc
```

Comment:
`__doc__` is a writable attribute, and descriptors in different namedtuples can have different docs. This is why itemgetter objects was cached instead property objects.
```python
A = namedtuple('A', 'x y')
B = namedtuple('B', 'x y')
A.x.__doc__ = 'foo'
B.x.__doc__ = 'bar'
assert A.x.__doc__ = 'foo'
assert B.x.__doc__ = 'bar'
```

Reply:
Ok, I will cache only the docstring then.

Reply:
I am caching only the docstring in 21be735